src/components/LibrariesListItem.tsx

```diff
@@ -0,0 +1,65 @@
+import * as React from "react";
+import Panel from "./reusables/Panel";
+import { Store } from "redux";
+import { State } from "../reducers/index";
+import { LibraryData } from "../interfaces";
+import LibraryDetailPage from "./LibraryDetailPage";
+import { GenericWedgeIcon } from "@nypl/dgx-svg-icons";
+
+export interface LibrariesListItemProps {
+  library: LibraryData;
+  store: Store<State>;
+}
+
+export interface LibrariesListItemState {
+  color: string;
```

Comment:
I'm not sure why this needs to be in the state. Is it so the color can update before the request to change it on the server?

Reply:
Exactly!

Reply:
This isn't great if the request causes an error - seems like the header color changes but the current stage and its color stay the same

Reply:
Agreed...but there's no error handling anywhere yet.  I figured that would be in the next phase.

Reply:
Ok--what do you think of this implementation?  Still using `state` in `LibrariesListItem`, but passing it values from the server.  (It seems like overkill to have the `LibrariesList` fetch all new data just for the sake of updating the header for one library...and also seems like overkill to connect `LibrariesListItem` to the store just to do this...right?)

Reply:
I'm not sure exactly what you mean about passing it values from the server. If the values are coming from the server, wouldn't they be in the props already?

One thing you could consider is to update the redux store after the edit request succeeds - we haven't used this pattern, but you can have an action that updates something in the store without making a request to the server. Maybe that would be cleaner?

Reply:
I mean I'm passing it new values from the server: the new data for this library, complete with the changes that the user has made via the form.  So that it re-renders with the new values for library_stage and registry_stage.  (See the long and semi-coherent essay that I wrote in response to Edwin's comment, at the bottom of the page!)

I'm not sure I understand how that pattern would work.  Do you mean, after the edit request succeeds, manually give the store the new library_stage and registry_stage values (so that the data in the store matches the data in the server, but doesn't actually come from the server)...?  At that point, why not just leave the store out of it and update the component's state with those values instead?  (Like what I had before, but waiting until after the edit request has succeeded.)

Reply:
This is exactly what I meant - thanks for explaining it better :)

Reply:
I agree about closing out this PR and experimenting with implementing this pattern in a different branch, given that the current implementation, while potentially not optimal, does actually work and doesn't seem to be hurting anything.  @aslagle, what do you think?

Reply:
@aslagle The more I think about it, I really really want to just close out this horrible giant PR and make further changes in separate branches.  Any strenuous objections?